### ISSUE TITLE
Hello author, I want to reproduce your single image degradation task in the experiment part, but I found that the validation set uses the same dataset as the training set, and val_freq is 1. Also, why do you set derain_ratio to 120? Do I need to modify additional parameters when training a single derain task? For example, weight decay is set to 1e-4 in the paper, but it is 1e-8 in the train.yml you provided?

### ISSUE THREAD

**Issue by Murmur007** [mdyao/NDR-Restore#6]
Hello author, I want to reproduce your single image degradation task in the experiment part, but I found that the validation set uses the same dataset as the training set, and val_freq is 1. Also, why do you set derain_ratio to 120? Do I need to modify additional parameters when training a single derain task? For example, weight decay is set to 1e-4 in the paper, but it is 1e-8 in the train.yml you provided?The train.yml file I modified is as follows, but the best PSNR test results of my trained model are only 25.8659 and the best SSIM is only 0.8108.
#### general settings
name: train
model: NDR
scale: 2
gpu_ids: [0]
#### datasets
datasets:
  train:
    name: Mix
    mode: NDR_train
    derain_gt: './example_dataset/train/rain/gt'
    derain_lq: './example_dataset/train/rain/rainy'
#  derain_ratio: 120
    derain_ratio: 1

    n_workers: 6  
    batch_size: 4
    GT_size: 128

  val:
    name: Mix_val
    mode: NDR_val
    dataroot_gt: './example_dataset/train/rain/gt'
    dataroot_lq: './example_dataset/train/rain/rainy'

#### network structures
network_G:
  which_model_G:
      subnet_type: Resnet
  in_nc: 3
  out_nc: 3
  block_num: [8, 8]
  scale: 2
  init: xavier

#### path

path:
  root: ./
  pretrain_model_G: 
  strict_load: false
  resume_state: 
#### training settings: learning rate scheme, loss

train:
  lr_G: !!float 1e-4
  beta1: 0.9
  beta2: 0.999
  niter: 4050
  warmup_iter: -1  # no warm up

  lr_scheme: MultiStepLR
  lr_steps: [200, 400, 600, 800]
  lr_gamma: 0.5

  pixel_criterion: l1

  manual_seed: 9

  # val_freq: !!float 1
  val_freq: !!float 200

  lambda_restore: 1
  lambda_degrad: 1

  # weight_decay_G: !!float 1e-8
  weight_decay_G: !!float 1e-4
  gradient_clipping: 10

#### logger

logger:
  print_freq: 100
  save_checkpoint_freq: !!float 1000

**Comment by mdyao** [comment]
Thank you for your interest in our work.  

Regarding the training and validation dataset paths, the train.yml file provided in our repository is intended as a sample configuration. We recommend setting the paths according to your specific dataset and experimental setup.  

Why is derain_ratio set to 120?  
Hi, to align with previous all-in-one methods (e.g., AirNet, PromptIR), in this repo, we provide model parameters trained using the same dataset ratio, enabling easy comparison with those methods. Specifically, the denoising dataset was replicated 3 times, the deraining dataset was replicated 120 times, and the dehazing dataset remained unchanged. Please refer to [https://github.com/va1shn9v/PromptIR/blob/main/utils/dataset_utils.py](url). Also refer to [#3](https://github.com/mdyao/NDR-Restore/issues/3#).

Weight decay:  
We recommend using a weight decay of 1e-8, as specified in the `train.yml` file.
